node/compile-hook.js

```diff
@@ -60,6 +60,7 @@ function extWrap(func, pkgInfo, mod, filePath) {
 
   // If the module is not loaded through module.run, then run its setters.
   if (! mod.loaded) {
+    mod.loaded = true;
```

Comment:
I can dig the explicit `loaded` being set here. (It's set by `Module.prototype.load` after this call but this clears things up.)